### PR TITLE
[Train] `HuggingFacePredictor` & docs improvements

### PR DESCRIPTION
**Pull request by Yard1**
## Why are these changes needed?

This PR introduces 2 changes:
* Removes a confusing suggestion in the docstring of `HuggingFaceCheckpoint` - checkpoints created using `from_checkpoint` will not work for prediction as intended.
* Adds `use_gpu` argument and logic to automatically use GPU if one is available to `HuggingFacePredictor`.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/train/tests/test_huggingface_gpu.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    with tempfile.TemporaryDirectory() as tmpdir:
+        model_config = AutoConfig.from_pretrained(model_checkpoint)
+        model = AutoModelForCausalLM.from_config(model_config)
+        tokenizer = AutoTokenizer.from_pretrained(tokenizer_checkpoint)
+        checkpoint = HuggingFaceCheckpoint.from_model(model, tokenizer, path=tmpdir)
+        # Serialize to dict so we can remove the temporary directory
+        return HuggingFaceCheckpoint.from_dict(checkpoint.to_dict())
+
+
+@pytest.mark.parametrize("batch_type", [pd.DataFrame])
```
[the remark this answers is not in the document]
Comment (Yard1) [reply python/ray/train/tests/test_huggingface_gpu.py:34]: this is the same as in `test_huggingface_predictor` - @jiaodong what's the reason for not testing other batch types with HuggingFace & BatchPredictor?

[the remark this answers is not in the document]
Comment (jiaodong) [reply python/ray/train/tests/test_huggingface_gpu.py:34]: i think previously some batch types we used had arrow and removed as part of numpy narrow waist work. For prediction we will only have `pandas` and `numpy` tho, as you can see the `pa.Table` option should no longer be applicable.

[the remark this answers is not in the document]
Comment (Yard1) [reply python/ray/train/tests/test_huggingface_gpu.py:34]: @amogkam hope that answers your question!

File: python/ray/train/tests/test_huggingface_gpu.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    predictor = BatchPredictor.from_checkpoint(
+        checkpoint, HuggingFacePredictor, task="text-generation"
+    )
+
+    # Todo: Ray data does not support numpy string arrays well
+    if batch_type == np.ndarray:
+        dataset = ray.data.from_numpy(prompts.to_numpy().astype("U"))
+    elif batch_type == pd.DataFrame:
+        dataset = ray.data.from_pandas(prompts)
+    elif batch_type == pa.Table:
```
Comment (amogkam) [inline python/ray/train/tests/test_huggingface_gpu.py:47]: let's remove arrow if it's no longer being supported?

File: python/ray/train/tests/test_huggingface_gpu.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (amogkam) [reply python/ray/train/tests/test_huggingface_gpu.py:34]: Thanks! Can we add a TODO here and in `test_huggingface_predictor.py` to re-enable the numpy test?

File: python/ray/train/tests/test_huggingface_gpu.py (hunk 2 above)
Comment (Yard1) [reply python/ray/train/tests/test_huggingface_gpu.py:47]: It's in multiple tests - we should do this in a separate PR

**Review by amogkam — APPROVED** [review]

**Pull request merged**